shared/lib_util.cpp

```diff
@@ -896,6 +896,20 @@ size_t util::hour_of_day(size_t hour_of_year)
 	return (hour_of_year) % 24;
 }
 
+//returns the hour index in the year (0-8759) of a given month, day, and time in hours
+size_t util::hour_of_year(size_t month, size_t day, size_t hour)
+{
+	size_t h = 0;
```

Comment:
Can we add an explicit check here that the input month is between 1 and 12, the input day is between 1 and days_in_month[month] and the input hour is between 0 and 23?

Reply:
Good suggestion! Done!